tokio-postgres/src/client.rs

```diff
@@ -310,16 +300,17 @@ impl Client {
         let stream = self.query_raw(statement, slice_iter(params)).await?;
         pin_mut!(stream);
 
-        let row = match stream.try_next().await? {
-            Some(row) => row,
-            None => return Ok(None),
-        };
+        let mut first = None;
+
+        while let Some(row) = stream.try_next().await? {
```

Comment:
Probably deserves a comment about why it's structured this way.

Reply:
Sure, added a comment.